pkg/fqdn/regexpmap/regexp_map_test.go

```diff
@@ -148,3 +160,38 @@ func (ds *RegexpMapTestSuite) BenchmarkRegexGroups(c *C) {
 		re.FindSubmatchIndex(in)
 	}
 }
+
+func RunRegexpMapBenchmark(c *C, matchAll bool) {
+	c.StopTimer()
+	iterations := 30000
+	m := NewRegexpMap()
+	matchformat := "foo%d.com."
+	lastMatch := 0
+
+	for i := 0; i < iterations; i++ {
+		key := fmt.Sprintf(matchformat, lastMatch)
+		val := fmt.Sprintf("ID%d", i)
+		m.Add(key, val)
+		if matchAll {
+			m.Add(".*", val)
+		}
+		if i%3 == 0 {
+			lastMatch = i
+		}
+	}
+	c.StartTimer()
+	for i := 0; i < iterations; i++ {
```

Comment:
This should use c.N, no? The numbers can't be correct since iterations is fixed.

Reply:
The main reason, Read is super fast, but c.N in write is super slow, so the test takes a lot of time to be relevant.

Reply:
But you can make a fixed size `RegexpMap` anyway and iterate over that c.N times (I'm guessing write here means building the RegexpMap). What does the benchmark measure when it gives you a time if `iterations` is a fixed value? Since iterations is fixed, changing c.N won't matter, so it will see different timings (since total_time/c.N is maller for bigger c.N). I must be missing something.